### ISSUE TITLE
💡 Feature Request - Support Azure Firewall Basic Tier

### ISSUE THREAD

**Issue by FallenHoot** [Azure/ALZ-Bicep#411]
## Describe the solution you'd like

[AB#25520](https://dev.azure.com/CSUSolEng/2a9d5a5a-8998-4ad0-81c8-ef3045e4da97/_workitems/edit/25520)

[Public Preview of Azure Firewall Basic](https://azure.microsoft.com/en-us/updates/public-preview-azure-firewall-basic/) has been out for a while. I understand that preview products can change, but it is 1/3 of the price for standard and works great for Small/Medium size companies, as well as PoCs. 

It requires the following code: 
parameters.all/min.json:
```
    "parSubnets": {
      "value": [
        {
          "name": "AzureFirewallManagementSubnet",
          "ipAddressRange": "10.20.255.0/24"
        }
      ]
    }
```

bicep:
```
resource resAzureFirewallManagementSubnetRef 'Microsoft.Network/virtualNetworks/subnets@2021-08-01' existing = {
  parent: resHubVnet
  name: 'AzureFirewallManagementSubnet'
}


// I didn't look into it too much, but don't think the below actually works for multiple Public IPs for the Firewall. Considering how the naming convention is.  
// AFW_Basic requires 2 PiP, but I simply copied and changed the naming.
module modAzureFirewallPublicIp2 '../publicIp/publicIp.bicep' = if (parAzFirewallEnabled) {
  name: 'deploy-Firewall-Public-IP2'
  params: {
    parLocation: parLocation
    parAvailabilityZones: parAzFirewallAvailabilityZones
    parPublicIpName: '${parAzFirewallName}-PublicIp2'
    parPublicIpProperties: {
      publicIpAddressVersion: 'IPv4'
      publicIpAllocationMethod: 'Static'
    }
    parPublicIpSku: {
      name: parPublicIpSku
    }
    parTags: parTags
    parTelemetryOptOut: parTelemetryOptOut
  }
}

// AzureFirewallManagementSubnet is required to deploy Azure Firewall Basic. This subnet must exist in the parsubnets array if you deploy.
// Two Public IPs are need for Azure Firewall Basic
resource resAzureFirewall 'Microsoft.Network/azureFirewalls@2021-08-01' = if (parAzFirewallEnabled) {
  name: parAzFirewallName
  location: parLocation
  tags: parTags
  zones: (!empty(parAzFirewallAvailabilityZones) ? parAzFirewallAvailabilityZones : [])
  properties: {
    ipConfigurations: [
      {
        name: 'ipconfig1'
        properties: {
          subnet: {
            id: resAzureFirewallSubnetRef.id
          }
          publicIPAddress: {
            id: parAzFirewallEnabled ? modAzureFirewallPublicIp.outputs.outPublicIpId : ''
          }
        }
      }
    ]
    sku: {
      name: 'AZFW_VNet'
      tier: parAzFirewallTier
    }
    managementIpConfiguration: {
      name: 'ManagementPublicIpAddressName'
      properties: {
        subnet: {
          id: resAzureFirewallManagementSubnetRef.id
        }
        publicIPAddress: {
          id: parAzFirewallEnabled ? modAzureFirewallPublicIp2.outputs.outPublicIpId : ''
        }
      }
    }
    firewallPolicy: {
      id: resFirewallPolicies.id
    }
  }
}

```

## Describe alternatives you've considered

Using Standard or Premium
Basic: $288.35 (Data Processing ($0.065) is not cheap, but takes 9.37 TB to hit Standard Tier pricing with 0 data processing)
Standard: $912.50 (Data Processing ($0.016))
Premium: $1,277.50 (Data Processing ($0.016))

## Additional context
![image](https://user-images.githubusercontent.com/19664186/205729693-438776b3-c48f-449f-8e39-2cf9fe20c88f.png)

![image](https://user-images.githubusercontent.com/19664186/205731234-f1c4c060-7cd6-483d-a37f-2eccf415bb23.png)

**Comment by jtracey93** [comment]
Thanks @FallenHoot for the request.

We already have this on our all up ALZ backlog to do in all 3 implementations.

I have linked this issue to the Bicep specific user story: ADO WIT ID: 25520

**Comment by spice67** [comment]
when is is the timeline on this?

**Comment by jtracey93** [comment]
This is available in the latest release v0.14.0
